MeshLib/Properties-impl.h

```diff
@@ -128,3 +128,33 @@ PropertyVector<T>* Properties::getPropertyVector(std::string const& name)
     return dynamic_cast<PropertyVector<T>*>(it->second);
 }
 
+template <typename T>
+PropertyVector<T> const* Properties::getPropertyVector(
+    std::string const& name, MeshItemType const item_type,
+    int const n_components) const
+{
+    auto const it = _properties.find(name);
+    if (it == _properties.end())
+    {
+        OGS_FATAL("A property with name `%s' does not exist.", name.c_str());
+    }
+
+    auto property = dynamic_cast<PropertyVector<T>*>(it->second);
+    if (property == nullptr)
+    {
+        OGS_FATAL("Could not cast property '%s' to given type.", name.c_str());
+    }
+    if (property->getMeshItemType() != item_type)
+    {
+        OGS_FATAL(
+            "Only nodal fields are supported for non-uniform fields. Field "
```

Comment:
The error message can be improved (maybe using function `toString(const MeshItemType t)` from `MeshLib::Location.h`).